virttest/qemu_monitor.py

```diff
@@ -282,7 +279,20 @@ def __init__(self, vm, name, filename, suppress_exceptions=False):
         self._supported_migrate_parameters = None
 
         try:
-            self._socket.connect(filename)
+            backend = monitor_params.get('chardev_backend', 'unix_socket')
+            if backend == 'tcp_socket':
+                self._socket = socket.socket(
+                    socket.AF_INET, socket.SOCK_STREAM)
+                self._socket.settimeout(self.CONNECT_TIMEOUT)
+                host = monitor_params['chardev_host']
+                port = int(monitor_params['chardev_port'])
+                self._socket.connect((host, port))
+            elif backend == 'unix_socket':
+                self._socket = socket.socket(
+                    socket.AF_UNIX, socket.SOCK_STREAM)
+                self._socket.settimeout(self.CONNECT_TIMEOUT)
+                file_name = monitor_params.get("monitor_filename")
+                self._socket.connect(file_name)
```

Comment:
I think here miss  else part like :

else:
       raise   Some Exception

Reply:
make sense, how to you think about raise some NotImplementedError exception ?

Reply:
`NotImplementedError` seems ok, please help to check, thanks.

Reply:
LGTM, @xutian  any other comment ?  thanks!